core/textile/sharing.go

```diff
@@ -142,86 +183,131 @@ func (tc *textileClient) GetReceivedFiles(ctx context.Context, accepted bool, se
 		return items, "", nil
 	}
 
+	var res *domain.SharedDirEntry
 	for _, file := range files {
-		ctx, _, err = tc.getBucketContext(ctx, file.DbID, file.Bucket, true, file.EncryptionKey)
-		if err != nil {
-			return nil, "", err
+		if file.IsPublicLinkReceived() {
+			res, err = tc.buildPublicLinkSharedDirEntry(ctx, file)
+		} else {
+			res, err = tc.buildInvitationSharedDirEntry(ctx, file)
 		}
 
-		sbc := NewSecureBucketsClient(tc.hb, file.Bucket)
-
-		f, err := sbc.ListPath(ctx, file.BucketKey, file.Path)
 		if err != nil {
 			return nil, "", err
 		}
+		items = append(items, res)
+	}
 
-		ipfsHash := f.Item.Cid
-		name := f.Item.Name
-		isDir := false
-		size := f.GetItem().Size
-		ext := strings.Replace(filepath.Ext(name), ".", "", -1)
+	offset := files[len(files)-1].ID.String()
```

Comment:
Will this offset work even when we are merging two entity lists?

Reply:
Yeah it should still work. We are not really merging two lists, it is only a single list being queried, but depending on the type of it's data we map it to a public shared file or a direct invite shared file.

Reply:
Understood 👍